src/services/refactors/extractMethod.ts

```diff
@@ -0,0 +1,885 @@
+/// <reference path="../refactorProvider.ts" />
+/// <reference path="../../compiler/checker.ts" />
+
+/* @internal */
+namespace ts.refactor.extractMethod {
+    export const name = "extract_method";
+    const extractMethod: Refactor = {
+        name: "Extract Method",
+        description: Diagnostics.Extract_function.message,
+        getAvailableActions,
+        getEditsForAction,
+    };
+
+    registerRefactor(extractMethod);
+
+    /** Compute the associated code actions */
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {

```

Comment:
It looks like this method returns at most one `ApplicableRefactorInfo`.  Why return an array?